ACEHAL/optimize_basis.py

```diff
@@ -247,7 +247,12 @@ def objective(trial):
         calc, Psi, Y, coef, _ = fit(fitting_db, solver, B_len_norm, return_linear_problem=True, **fit_kwargs_use)
 
         n = Psi.shape[0]
-        k = Psi.shape[1]
+
+        if hasattr(solver, 'threshold_lambda'):
```

Comment:
If you just did `try` / `catch`, you would also fall back to sensible behavior for solvers that don't have `lambda_`.

Reply:
The safest thing might actually be to explicitly check `isinstance(solver, sklearn.linear_solver.ARDRegression)`

Reply:
Responding to your slack comment, you could do
```
try:
    from sklearn.linear_solver import ARDRegression as sklearn_ARDRegression
except ModuleNotFoundError:
    sklearn_ARDRegression = None
.
.
.
if sklearn_ARDRegression is not None and isinstance(solver, sklearn_ARDRegression):
.
.
```

Or we could just to try/catch instead.  It _is_ the pythonic way, although in this case I find it slightly dangerous.

Reply:
I guess it's OK to assume that no one will define `solver.lambda_` in an inconsistent way.

Reply:
I just went with the sklearn dependency, I think you added it as a dependency too?

Reply:
I sort of agree that it shouldn't be a dependency, though, so I'd rather have this part working even if it's not available.  I'm going to fix that (remove the dependency from ACEHAL itself, just install it as part of the pytest because the tests use those solvers for the moment). in the github CI pytest PR.